component/http/handler.go

```diff
@@ -62,44 +62,76 @@ func determineEncoding(r *http.Request) (string, encoding.DecodeFunc, encoding.E
 	var enc encoding.EncodeFunc
 	var dec encoding.DecodeFunc
 	var ct string
+	var foundValidHeader bool
 
 	if cok {
-		switch cth[0] {
-		case "*/*", json.Type, json.TypeCharset:
-			enc = json.Encode
-			dec = json.Decode
-			ct = json.TypeCharset
-		case protobuf.Type, protobuf.TypeGoogle:
-			enc = protobuf.Encode
-			dec = protobuf.Decode
-			ct = protobuf.Type
-		default:
+		foundValidHeader = false
+		ah := getMultiValueHeaders(cth[0])
+		for _, v := range ah {
+			switch v {
+			case "*/*", json.Type, json.TypeCharset:
```

Comment:
With the above for loop the last match wins. Maybe we should break when the first match happens.

Reply:
Alright I pushed a fix but I am not fond of named loops, I could also get rid of switch and have an if/else if now that we don't have the default anymore. Whatever you like more I don't have strong feelings.

Reply:
I also do not like of labels.  You can actually get rid of them just by putting this code inside a new function and return the values set...

Reply:
Sounds good, will try that